dglc/dglc_datamode_noevolve_mod.F90

```diff
@@ -361,31 +370,105 @@ subroutine dglc_datamode_noevolve_advance(pio_subsystem, io_type, io_format, &
             end do
 
             deallocate(lsrf)
-
+            deallocate(usrf)
             call pio_closefile(pioid)
             call pio_freedecomp(pio_subsystem, pio_iodesc)
 
          end do ! end loop over ice sheets
 
       end if
 
+      ! Set initialized flag
+      initialized_noevolve = .true.
+      
       if (initialized_noevolve) then
+
          ! Compute Fgrg_rofi
          do ns = 1,num_icesheets
-            lsize = size(Fgrg_rofi(ns)%ptr)
+
+            ! Get mesh info
+            call ESMF_MeshGet(model_meshes(ns), elementdistGrid=distGrid, rc=rc)
+            if (ChkErr(rc,__LINE__,u_FILE_u)) return
+            call ESMF_DistGridGet(distGrid, localDe=0, elementCount=lsize, rc=rc)
+            if (ChkErr(rc,__LINE__,u_FILE_u)) return
```

Comment:
Is there a reason you repeat this fetching of lsize rather than using the original `lsize = size(Fgrg_rofi(ns)%ptr)`?

Reply:
I agree with @billsacks - I think that lsize should be set as a module variable on initialization.

Reply:
I'm just not clear on everything that is done in these two function calls, but I removed the second one and replaced it with the size(Fgrg_rofi(ns)%ptr) function instead as that does seem clear and simple. I disagree with Mariana that lsize should be a module variable, as I think that would require writing it out for restarts, and that's unnecessary.

Reply:
> I'm just not clear on everything that is done in these two function calls

The ESMF_MeshGet call says: I've got this mesh (`model_meshes(ns)`), give me the associated elementdistGrid (`distGrid`); then ESMF_DistGridGet says: I've got this distgrid, give me the elementCount (`lsize`). (This is a common ESMF pattern: I've got this object, I want to get one or more of its components or other related information, so call a `Get` function that takes that object as an argument and various optional output arguments.)